protocol/message.go

```diff
@@ -0,0 +1,50 @@
+package protocol
+
+import "github.com/coniks-sys/coniks-go/merkletree"
+
+// Defines constants representing the types
+// of messages exchanged by clients and servers.
+const (
+	RegistrationType = 0
+	CommitmentType   = 1
```

Comment:
> But I don't think there's a separate STR message in this implementation, at least for now, right?

How's about the monitoring function? Does the client still need to obtain some missed STRs to verify the hash chain anyways? Maybe I can rename it to `ConsistencyCheckType`?